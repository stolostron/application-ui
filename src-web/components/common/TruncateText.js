/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import React from 'react'
import PropTypes from 'prop-types'
import { Tooltip } from 'carbon-components-react'
import resources from '../../../lib/shared/resources'
import truncate from '../../util/truncate-middle'

resources(() => {
  require('../../../scss/textWithTruncation.scss')
})

class TruncateText extends React.PureComponent {
  static propTypes = {
    maxCharacters: PropTypes.number,
    text: PropTypes.string.isRequired
  };

  static defaultProps = {
    maxCharacters: 35
  };

  render() {
    if (this.props.text.length <= this.props.maxCharacters) {
      return this.props.text
    }

    return (
      <Tooltip
        triggerClassName="textWithTruncation"
        showIcon={false}
        text={this.props.text}
        triggerText={truncate(this.props.text, this.props.maxCharacters)}
      >
        {this.props.text}
      </Tooltip>
    )
  }
}

export default TruncateText
