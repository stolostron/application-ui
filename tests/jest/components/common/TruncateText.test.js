/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
"use strict";

import React from "react";
import TruncateText from "../../../../src-web/components/common/TruncateText";
import renderer from "react-test-renderer";

describe("TruncateText", () => {
  it("renders as expected for truncated text", () => {
    const props = { maxCharacters: 7, text: "TruncateText" };
    const component = renderer.create(<TruncateText {...props} />);
    expect(component.getInstance().render().props).toMatchSnapshot();
  });

  it("renders as expected for non truncated text", () => {
    const props = { maxCharacters: 25, text: "NonTruncateText" };
    const component = renderer.create(<TruncateText {...props} />);
    expect(component).toMatchSnapshot();
  });
});
