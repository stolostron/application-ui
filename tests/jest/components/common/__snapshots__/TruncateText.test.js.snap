// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruncateText renders as expected for non truncated text 1`] = `"NonTruncateText"`;

exports[`TruncateText renders as expected for truncated text 1`] = `
Object {
  "children": "TruncateText",
  "direction": "bottom",
  "iconDescription": "tooltip",
  "iconTitle": "",
  "menuOffset": [Function],
  "open": false,
  "showIcon": false,
  "text": "TruncateText",
  "triggerClassName": "textWithTruncation",
  "triggerText": "Tr...xt",
}
`;
